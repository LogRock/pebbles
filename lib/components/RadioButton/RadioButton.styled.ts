import styled from "styled-components";

export const RadioButtonContainer = styled.div`
  display: flex;
  flex-flow: row nowrap;
  align-items: flex-start;
  margin-top: ${({ theme }) => theme.spacings.sm};
  margin-bottom: ${({ theme }) => theme.spacings.sm};
`;

export const RadioButtonInput = styled.input`
  width: 0;
  height: 0;
`;

export const RadioButtonRadio = styled.div<{ checked?: boolean }>`
  position: relative;
  box-sizing: border-box;
  width: ${({ theme }) => theme.spacings.md};
  height: ${({ theme }) => theme.spacings.md};
  margin-right: ${({ theme }) => theme.spacings.sm};

  border: 1.5px solid
    ${({ theme, checked }) =>
      checked ? theme.colors.primary[500] : theme.colors.neutral[300]};
  border-radius: ${({ theme }) => theme.spacings.md};
`;

export const RadioButtonMarker = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  box-sizing: border-box;
  width: ${({ theme }) => theme.spacings.xsm};
  height: ${({ theme }) => theme.spacings.xsm};

  transform: translate(-50%, -50%);

  border-radius: ${({ theme }) => theme.spacings.xsm};

  background-color: ${({ theme }) => theme.colors.primary[500]};
`;
