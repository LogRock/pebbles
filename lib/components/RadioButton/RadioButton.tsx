import uniqueId from "lodash.uniqueid";
import React, { FC, InputHTMLAttributes } from "react";
import {
  RadioButtonContainer,
  RadioButtonInput,
  RadioButtonMarker,
  RadioButtonRadio,
} from "./RadioButton.styled";

const RadioButton: FC<InputHTMLAttributes<HTMLInputElement>> = ({
  children,
  checked,
  id,
  ...props
}) => {
  const newID = id || uniqueId("logrock_pebbles_radiobutton");

  return (
    <RadioButtonContainer>
      <RadioButtonInput
        type="radio"
        {...props}
        id={newID}
        checked={checked}
      ></RadioButtonInput>
      <RadioButtonRadio checked={checked}>
        {checked && <RadioButtonMarker />}
      </RadioButtonRadio>
      <label htmlFor={newID}>{children}</label>
    </RadioButtonContainer>
  );
};

export default RadioButton;
