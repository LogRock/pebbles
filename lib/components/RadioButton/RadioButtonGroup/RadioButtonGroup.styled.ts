import styled from "styled-components";
import { overlineXSmallCSS } from "../../Typography";

export const RadioButtonGroupLabel = styled.div`
  ${overlineXSmallCSS}
  margin-bottom: ${({ theme }) => theme.spacings.xxsm};
`;
