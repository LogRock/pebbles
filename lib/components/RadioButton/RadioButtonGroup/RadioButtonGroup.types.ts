import React, { HTMLProps } from "react";

export interface RadioButtonGroupProps extends HTMLProps<HTMLDivElement> {
  groupTitle?: React.ReactNode;
}
