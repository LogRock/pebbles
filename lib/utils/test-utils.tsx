/* eslint-disable import/export */
import React from "react";
import { cleanup, render } from "@testing-library/react";
import { afterEach } from "vitest";
import { ThemeProvider } from "styled-components";
import { light as theme } from "../theme";

afterEach(() => {
  cleanup();
});

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
// override render export
export { customRender as render };
