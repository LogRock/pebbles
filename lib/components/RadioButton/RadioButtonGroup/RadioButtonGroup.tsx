import React, { FC } from "react";
import { RadioButtonGroupLabel } from "./RadioButtonGroup.styled";
import { RadioButtonGroupProps } from "./RadioButtonGroup.types";

const RadioButtonGroup: FC<RadioButtonGroupProps> = (props) => {
  return (
    <div {...props}>
      <RadioButtonGroupLabel>{props.groupTitle}</RadioButtonGroupLabel>
      {props.children}
    </div>
  );
};

export default RadioButtonGroup;
