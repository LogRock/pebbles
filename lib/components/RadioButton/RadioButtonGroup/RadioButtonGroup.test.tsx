import React from "react";
import RadioButtonGroup from ".";
import RadioButton from "../";
import { render, screen } from "../../../utils/test-utils";

describe("RadioButtonGroup", () => {
  it("render properly", () => {
    const { container } = render(
      <RadioButtonGroup groupTitle="Pick at least 6 options">
        <RadioButton>First option</RadioButton>
        <RadioButton>Option B</RadioButton>
        <RadioButton>#3</RadioButton>
        <RadioButton>Last one</RadioButton>
        <RadioButton>Option the V</RadioButton>
      </RadioButtonGroup>
    );

    expect(screen.getByText("Pick at least 6 options")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
