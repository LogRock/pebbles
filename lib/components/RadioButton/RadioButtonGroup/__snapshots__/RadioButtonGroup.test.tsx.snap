// Vitest Snapshot v1

exports[`RadioButtonGroup > render properly 1`] = `
<div
  grouptitle="Pick at least 6 options"
>
  <div
    class="sc-bqiRlB hnwbD"
  >
    Pick at least 6 options
  </div>
  <div
    class="sc-fFeiMQ gZqCdu"
  >
    <input
      class="sc-ksdxgE gRYfSo"
      id="logrock_pebbles_radiobutton1"
      type="radio"
    />
    <div
      class="sc-hBUSln clbhns"
    >
      <div
        class="sc-fotOHu eMZCAq"
      />
    </div>
    <label
      for="logrock_pebbles_radiobutton1"
    >
      First option
    </label>
  </div>
  <div
    class="sc-fFeiMQ gZqCdu"
  >
    <input
      class="sc-ksdxgE gRYfSo"
      id="logrock_pebbles_radiobutton2"
      type="radio"
    />
    <div
      class="sc-hBUSln clbhns"
    >
      <div
        class="sc-fotOHu eMZCAq"
      />
    </div>
    <label
      for="logrock_pebbles_radiobutton2"
    >
      Option B
    </label>
  </div>
  <div
    class="sc-fFeiMQ gZqCdu"
  >
    <input
      class="sc-ksdxgE gRYfSo"
      id="logrock_pebbles_radiobutton3"
      type="radio"
    />
    <div
      class="sc-hBUSln clbhns"
    >
      <div
        class="sc-fotOHu eMZCAq"
      />
    </div>
    <label
      for="logrock_pebbles_radiobutton3"
    >
      #3
    </label>
  </div>
  <div
    class="sc-fFeiMQ gZqCdu"
  >
    <input
      class="sc-ksdxgE gRYfSo"
      id="logrock_pebbles_radiobutton4"
      type="radio"
    />
    <div
      class="sc-hBUSln clbhns"
    >
      <div
        class="sc-fotOHu eMZCAq"
      />
    </div>
    <label
      for="logrock_pebbles_radiobutton4"
    >
      Last one
    </label>
  </div>
  <div
    class="sc-fFeiMQ gZqCdu"
  >
    <input
      class="sc-ksdxgE gRYfSo"
      id="logrock_pebbles_radiobutton5"
      type="radio"
    />
    <div
      class="sc-hBUSln clbhns"
    >
      <div
        class="sc-fotOHu eMZCAq"
      />
    </div>
    <label
      for="logrock_pebbles_radiobutton5"
    >
      Option the V
    </label>
  </div>
</div>
`;
